src/ratio.jl

```diff
@@ -0,0 +1,31 @@
+# Base's Rational type is too slow because of its penchant for calling gcd and div.
+# So we roll our own.
```

Comment:
Without having perused the code carefully, this seems like something we shouldn't have to do; if for no other reason, then because there is probably also other packages and code bases that could benefit from this speed improvements.

In the long term, maybe we should split these out and incorporate them either in Base or in their own package? (I imagine these do a subset of the stuff that `Base.Rational` do - maybe we could re-write `Rational`s to be a two-step rocket, exposing these too? Loong-term goal...)